src/components/Forms/Health/MongoHealthFormEditor.tsx

```diff
@@ -0,0 +1,72 @@
+import FormikTextInput from "../Formik/FormikTextInput";
+import FormikTemplateFields from "../Formik/FormikTemplateFields";
+import FormikAuthFieldsGroup from "../Formik/FormikAuthFieldsGroup";
+import FormikCheckboxFieldsGroup from "../Formik/FormikCheckboxFieldsGroup";
+import FormikIconPicker from "../Formik/FormikIconPicker";
+import { getIn, useFormikContext } from "formik";
+import { useEffect } from "react";
+import FormikScheduleField from "../Formik/FormikScheduleField";
+
+type MongoHealthFormEditorProps = {
+  fieldName: string;
+  specsMapField: string;
+};
+
+export function MongoHealthFormEditor({
+  fieldName: name,
+  specsMapField
+}: MongoHealthFormEditorProps) {
+  const { values, setFieldValue } = useFormikContext();
+
+  const fieldName = `${name}.${specsMapField}`;
+
+  const nameValue = getIn(values, `${fieldName}.name`);
+
+  // when name changes, we want to update the name of the top level field
+  useEffect(() => {
+    setFieldValue("name", nameValue);
+  }, [nameValue, setFieldValue]);
+
+  return (
+    <>
+      <div className="flex flex-row md:space-x-2">
+        <FormikTextInput
+          className="flex flex-col w-1/2"
+          name={`${fieldName}.name`}
+          label="Name"
+          required
+        />
+        <FormikIconPicker
+          className="flex flex-col w-1/2"
+          name={`${fieldName}.icon`}
+          label="Icon"
+        />
+      </div>
+
+      <FormikScheduleField name={`${name}.schedule`} />
+
+      <FormikTextInput name={`${fieldName}.username`} label="Username" />
+      <FormikTextInput name={`${fieldName}.password`} label="Password" />
```

Comment:
These should be env var fields + we need a connection selection field

Reply:
Also missing the url field

Reply:
@moshloop for the connection field, is it a dropdown connection to the connection view?